app/backend/src/couchers/models.py

```diff
@@ -179,9 +179,14 @@ class User(Base):
 
     # for changing their email
     new_email = Column(String, nullable=True)
+    old_email_token = Column(String, nullable=True)
+    old_email_token_created = Column(DateTime(timezone=True), nullable=True)
+    old_email_token_expiry = Column(DateTime(timezone=True), nullable=True)
+    confirmed_email_change_via_old_email = Column(Boolean, nullable=True, default=False)
     new_email_token = Column(String, nullable=True)
     new_email_token_created = Column(DateTime(timezone=True), nullable=True)
     new_email_token_expiry = Column(DateTime(timezone=True), nullable=True)
+    confirmed_email_change_via_new_email = Column(Boolean, nullable=True, default=False)
```

Comment:
I think given your usage here, `confirmed_email_change_via_{new,old}_email` would be better named in the opposite, `need_to_confirm_{new,old}_email`.

Also; there's a lot of fields here that need to work in tandem: could you add a constraint to make sure that `old_email_*,confirmed_email_change_via_old_email` are all either or all NULL, and similar for `new_email_*,...`?

Reply:
I can add constraints that old_email_* are all NULL or all not NULL (I think that's what you want?), but as it's currently set up, confirmed_email_change_via_old_email can be True while they are NULL (i.e. user has confirmed via old email but not yet through new email). I could change it so that even once it's confirmed the fields don't go to null until the whole process is complete. I suppose it doesn't really matter? Worst case scenario they just confirm the email change again using the same email/token and nothing really changes?